processor/metricstransformprocessor/factory.go

```diff
@@ -128,7 +166,20 @@ func buildHelperConfig(config *Config, version string) []internalTransform {
 		}
 		helperDataTransforms[i] = helperT
 	}
-	return helperDataTransforms
+	return helperDataTransforms, nil
+}
+
+func createFilter(filterConfig FilterConfig) (internalFilter, error) {
+	switch filterConfig.MatchType {
+	case RegexpMatchType:
+		r, err := regexp.Compile(filterConfig.Include)
+		if err != nil {
+			return nil, err
+		}
+		return internalFilterRegexp{include: r}, nil
+	default:
+		return internalFilterStrict{include: filterConfig.Include}, nil
```

Comment:
Validation actually occurs before this function runs so that shouldn't be possible. Nevertheless, I removed the `default` case so invalid values will result in an error instead of silently using `Strict`.